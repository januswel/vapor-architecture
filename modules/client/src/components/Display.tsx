import * as React from 'react'
import { FlatList, StyleSheet } from 'react-native'

import Item, { Props as ItemProps, Actions as ItemActions } from './Item'

const styles = StyleSheet.create({
  container: {
    marginTop: 16,
  },
})

export type Items = Array<ItemProps>
export type Actions = ItemActions
export interface Props {
  items: Items
  chargedMoney: number
  actions: Actions
}
export default (props: Props) => (
  <FlatList
    style={styles.container}
    data={props.items}
    extraData={props.chargedMoney}
    numColumns={2}
    renderItem={({ item }) => <Item {...item} actions={props.actions} />}
    keyExtractor={item => item.id.toString(10)}
  />
)
