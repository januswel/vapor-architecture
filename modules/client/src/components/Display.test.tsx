import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Display from './Display'

describe('Display', () => {
  it('renders display which have no items', () => {
    const component = renderer.create(
      <Display
        items={[]}
        chargedMoney={0}
        actions={{
          sellAndCount: () => {},
        }}
      />,
    )
    expect(component).toMatchSnapshot()
  })

  it('renders display which have some items', () => {
    const component = renderer.create(
      <Display
        items={[
          {
            id: 0,
            price: 120,
            imageUrl: 'https://example.com/assets/images/coffee.png',
            isSoldOut: false,
            isTooFewChargedMoney: true,
            actions: {
              sellAndCount: () => {},
            },
          },
          {
            id: 1,
            price: 100,
            imageUrl: 'https://example.com/assets/images/water.png',
            isSoldOut: true,
            isTooFewChargedMoney: true,
            actions: {
              sellAndCount: () => {},
            },
          },
        ]}
        chargedMoney={0}
        actions={{
          sellAndCount: () => {},
        }}
      />,
    )
    expect(component).toMatchSnapshot()
  })
})
