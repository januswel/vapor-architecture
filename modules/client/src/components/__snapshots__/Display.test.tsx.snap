// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders display which have no items 1`] = `
<RCTScrollView
  data={Array []}
  disableVirtualization={false}
  extraData={0}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={2}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginTop": 16,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;

exports[`Display renders display which have some items 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "actions": Object {
          "sellAndCount": [Function],
        },
        "id": 0,
        "imageUrl": "https://example.com/assets/images/coffee.png",
        "isSoldOut": false,
        "isTooFewChargedMoney": true,
        "price": 120,
      },
      Object {
        "actions": Object {
          "sellAndCount": [Function],
        },
        "id": 1,
        "imageUrl": "https://example.com/assets/images/water.png",
        "isSoldOut": true,
        "isTooFewChargedMoney": true,
        "price": 100,
      },
    ]
  }
  disableVirtualization={false}
  extraData={0}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={2}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginTop": 16,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 168,
                "justifyContent": "space-around",
                "margin": 16,
                "width": 168,
              },
              Object {
                "backgroundColor": "gray",
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "uri": "https://example.com/assets/images/coffee.png",
              }
            }
            style={
              Object {
                "height": 96,
                "width": 96,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            ¥
            120
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 168,
                "justifyContent": "space-around",
                "margin": 16,
                "width": 168,
              },
              Object {
                "backgroundColor": "red",
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "uri": "https://example.com/assets/images/water.png",
              }
            }
            style={
              Object {
                "height": 96,
                "width": 96,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            ¥
            100
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
